core/application/utils/geohash.ts

```diff
@@ -16,3 +17,62 @@ export function decodeGeohash(geohash: string): { latitude: number; longitude: n
     return null
   }
 }
+
+function getNeighbors(geohash: string): string[] {
+  return ngeohash.neighbors(geohash)
+}
+
+export function getGeohashNthNeighbors(geohash: string, neighborLevel: number): string[] {
```

Comment:
```ts
export function getGeohashNthNeighbors(geohash: string, neighborLevel: number): string[] {
  const searchedHashes = new Set([geohash])
  const hashesToSearch = [{ hash: geohash, level: 0 }]

  const nthLevelNeighbors: string[] = []

  while (hashesToSearch.length > 0) {
    const { hash, level } = hashesToSearch.shift()!

    if (level === neighborLevel) {
      nthLevelNeighbors.push(hash)
    }

    if (level < neighborLevel) {
      for (const neighbor of getNeighbors(hash)) {
        if (!searchedHashes.has(neighbor)) {
          searchedHashes.add(neighbor)
          hashesToSearch.push({ hash: neighbor, level: level + 1 })
        }
      }
    }
  }

  return nthLevelNeighbors
}
```

would this be cleaner?

Reply:
Updated in 6d0a4e676be617fafddc39c5ecaf64fd0bde4043